woocommerce/woocommerce
# Add subscribe notices for products without subscription

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Fixes https://github.com/Automattic/woocommerce.com/issues/21455

Add a new notice to the plugins list for products used without subscriptions reminding them to purchase.

![CleanShot 2024-09-04 at 09 45 56@2x](https://github.com/user-attachments/assets/b608115f-a417-45b2-a435-20f77aae43f7)


[As described here](https://arc.net/l/quote/viyavoce), this also removes notices for expired and expiring subscriptions from WooCommerce settings screens.

Add a new notice to the WooCommerce screens for missing subscriptions:

![CleanShot 2024-09-04 at 09 45 27@2x](https://github.com/user-attachments/assets/9bb880fb-7c59-4676-95a7-48ace074396c)
![CleanShot 2024-09-02 at 10 25 44@2x](https://github.com/user-attachments/assets/3efed41c-2e6d-4125-8ecc-4b3a4a3ef884)

This is part of the PU2 project, [project thread](https://woomarketplace.wordpress.com/2024/06/20/project-thread-prevent-usage-without-active-subscription-phase-2/).



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Connect your WC Core site to woocommerce.test [as described in this guide](https://wp.me/p6JqRr-cDo).
2. Make sure you have some purchased subscriptions in woocommerce.test.
3. Install a plugin without an active subscription, and make sure that you see the notices above in the plugins list and the WooCommerce screens.
4. Make sure that the notices for expired and expiring subscriptions remain unchanged.
![CleanShot 2024-08-30 at 15 01 49@2x](https://github.com/user-attachments/assets/7b5e33c2-cdc5-4dbc-af63-71693fb45f7e)
5. The notices for missing subscriptions in WooCommerce screens should only show up if you don't have any expired or expiring subscriptions. i.e. missing subscriptions have the lowest priority.
![CleanShot 2024-09-02 at 10 36 37@2x](https://github.com/user-attachments/assets/c7b7c521-c187-4272-8566-c035665dddb2)
7. Make sure that expiring and expired subscriptions are not visible in the settings screens., i.e. this should not appear:
![CleanShot 2024-09-02 at 10 39 58@2x](https://github.com/user-attachments/assets/f2308a07-6c4b-4af5-a4b2-9a75db0e1b95)
8. For missing and expired subscriptions, the link at both the plugin screen and extensions screen should point to the add to cart page. Check this both with a single product and multiple products. Click on the link and ensure that products are added to cart:
![CleanShot 2024-09-11 at 11 27 21@2x](https://github.com/user-attachments/assets/81b73f2e-ffc4-4494-9698-1c98a4d086fe)
![CleanShot 2024-09-11 at 11 28 33@2x](https://github.com/user-attachments/assets/f9b4bb92-3ad1-4de0-b1d0-9747f696c724)
8. Please test for any edge cases.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add a new notice to the plugins list for products used without subscriptions reminding them to purchase.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
